# Introduccion

El dataset del Titanic es uno de los conjuntos de datos más conocidos y ampliamente utilizados en ciencia de datos y aprendizaje automático. Este dataset contiene información sobre los pasajeros del Titanic, el famoso transatlántico que se hundió en su viaje inaugural en 1912

![](../img/header.png)

## Descripcion de los datos

| Columna      | Descripción                                         | Notas                                                                                       |
|--------------|-----------------------------------------------------|---------------------------------------------------------------------------------------------|
| PassengerId  | Identificador único para cada pasajero              |                                                                                             |
| Survived     | Indica si el pasajero sobrevivió (1) o no (0)       |                                                                                             |
| Pclass       | Clase del boleto del pasajero (1 = Primera, 2 = Segunda, 3 = Tercera) | Una proxy para el estatus socioeconómico (SES). 1st = Alta, 2nd = Media, 3rd = Baja        |
| Name         | Nombre del pasajero                                 |                                                                                             |
| Sex          | Género del pasajero                                 |                                                                                             |
| Age          | Edad del pasajero                                   | La edad es fraccional si es menor a 1. Si la edad es estimada, está en la forma de xx.     |
| SibSp        | Número de hermanos y cónyuges a bordo del Titanic   | El dataset define las relaciones familiares de esta manera: Hermano = hermano, hermana, hermanastro, hermanastra. Cónyuge = esposo, esposa (se ignoraron amantes y prometidos) |
| Parch        | Número de padres e hijos a bordo del Titanic        | El dataset define las relaciones familiares de esta manera: Padre = madre, padre. Hijo = hija, hijo, hijastra, hijastro. Algunos niños viajaban solo con una niñera, por lo tanto parch=0 para ellos. |
| Ticket       | Número del boleto                                   |                                                                                             |
| Fare         | Tarifa pagada por el boleto                         |                                                                                             |
| Cabin        | Número de cabina                                    |                                                                                             |
| Embarked     | Puerto de embarque (C = Cherbourg, Q = Queenstown, S = Southampton) |                                                                                             |


## Carga de datos

In [1]:
import pandas as pd 

train_df = pd.read_csv('../data/train.csv')
test_df = pd.read_csv('../data/test.csv')

In [2]:
# Mostrar las primeras filas del dataset de entrenamiento
train_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [3]:
# Mostrar las primeras filas del dataset de prueba
test_df.head()

PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

**Es importante resaltar que el conjundo de datos prueba no tiene la columna Survived(Supervivencia) debido a que sera la columna objetivo para determinar el modelo de prediccion**

## Exploracion de datos

In [4]:
# Información general del dataset de entrenamiento
print("Información general del dataset:")
print(train_df.info())

Información general del dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
None


* Las columnas **PassengerId, Survived, Pclass, Name, Sex, SibSp, Parch, Ticket, y Fare** no tienen valores nulos.
* La columna **Age** tiene 714 valores no nulos, lo que significa que faltan 177 valores.
* La columna **Cabin** tiene 204 valores no nulos, lo que indica que faltan 687 valores.
* La columna **Embarked** tiene 889 valores no nulos, con 2 valores faltantes.

In [5]:
# Estadísticas descriptivas
print("\nEstadísticas descriptivas:")
print(train_df.describe())


Estadísticas descriptivas:
       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  


In [6]:
# Valores nulos
print("\nValores nulos por columna:")
print(train_df.isnull().sum())


Valores nulos por columna:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


## Consideraciones iniciales importantes:

1. **Valores Faltantes**:
   - La columna **Age** tiene 714 valores no nulos, indicando que faltan datos para 177 pasajeros.
   - La columna **Cabin** tiene una gran cantidad de valores faltantes (687 valores faltantes).

2. **Distribución de Supervivencia**:
   - La media de la columna **Survived** es 0.3838, lo que sugiere que aproximadamente el 38.38% de los pasajeros sobrevivieron.

3. **Distribución de la Clase**:
   - La mayoría de los pasajeros están en la tercera clase, ya que la mediana (50%) y la moda de la columna **Pclass** es 3.

4. **Edad**:
   - La media de la edad es de aproximadamente 29.7 años, con una desviación estándar de 14.5 años, indicando una amplia distribución de edades.
   - El mínimo valor de edad es 0.42 años (menos de 1 año) y el máximo es 80 años.

5. **Tarifa**:
   - La tarifa tiene una gran variabilidad, con una desviación estándar alta (49.69), y la tarifa máxima es considerablemente alta (512.33).

6. **Familiares a bordo**:
   - La mayoría de los pasajeros no viajaban con hermanos/cónyuges (media de SibSp es 0.523) ni con padres/hijos (media de Parch es 0.3816).
